clappr/clappr-ios
# Sends cookies associated with a domain in AVURLAsset

## Goal

To send cookies associated with a domain when instantiating an `AVURLAsset`.

## How to test

Replace `viewDidLoad()` in `Example/ViewController.swift` code for this:

```swift
    override func viewDidLoad() {
        super.viewDidLoad()

        let cookie = HTTPCookie(properties: [HTTPCookiePropertyKey.domain: "clappr.io",
                                HTTPCookiePropertyKey.path: "/",
                                HTTPCookiePropertyKey.name: "testing",
                                HTTPCookiePropertyKey.value: "it was sent"])!
        HTTPCookieStorage.shared.setCookie(cookie)

        player = Player(options: options)

        listenToPlayerEvents()

        player.attachTo(playerContainer, controller: self)
    }
```

This code will set a cookie for the domain `clappr.io`. Then, play the video and check that the cookie was sent (using Charles, for example):

![screen shot 2017-11-06 at 14 10 06](https://user-images.githubusercontent.com/7040243/32450904-a06bdd9a-c2fc-11e7-8320-b4d1ac065630.png)
